ccmlib/common.py

```diff
@@ -241,6 +242,18 @@ def make_cassandra_env(install_dir, node_path, update_conf=True):
     env['cassandra.config'] = "file://" + os.path.join(node_path, 'conf', 'cassandra.yaml')
     env['CASSANDRA_HOME'] = install_dir
     env['CASSANDRA_CONF'] = os.path.join(node_path, 'conf')
+
+    if hardcode_java_home_8:
+        output = subprocess.check_output(["update-alternatives", "--display", "java"], text=True).splitlines()
+        output = [l.split()[0] for l in output if l.startswith('/')]
+        for java_path in output:
+            known_jvm8_names = ['8-openjdk', '8-jdk']
+            if any([name in java_path for name in known_jvm8_names]):
```

Comment:
i'd suggest change the parameter to `java_version`, which defaults to `1.8.0`, and put something like
```py
if java_version is not None:
    java_home = f"/usr/lib/jvm/java-{java_version}"
    java = shutil.which("java', path=os.path.join(java_home, 'bin'))
    if java is None:
        raise ArgumentError(f"java-8 wasn't found in alternatives: {output}")
    env['JAVA_HOME'] = java_home
```

Reply:
each env we run this had totally different names, there are multiple `java_version` we should be looking.

Reply:
i see. the suggestion was posted to explain the possibility to trade a hardwired boolean to a more flexible variable, if we want to test multiple, we can use a list instead of a single value here. and even your implementation only test for a single java version, right?

Reply:
supporting multiple versions, would need much more work then this.

the whole thing I'm trying to avoid using JAVA_HOME, since it would affect also other tools.

I'll open up a an issue for ccm, so we can support both 3.11 and 4.0 and up

Reply:
we want to support also ubuntu, and running github actions (which we use something else)

using `subprocess.check_output` vs. `shutil.which` isn't big of a difference from my POV

Reply:
> > supporting multiple versions, would need much more work then this.
> > the whole thing I'm trying to avoid using JAVA_HOME, since it would affect also other tools.
> 
> i am confused. you are actually using it. and it only affect the process which consumes the return value of `make_cassandra_env()`.

`env` is passed down directly to the process of cassandra, setting JAVA_HOME by the user/test would affect all process after, like scylla jmx processes.

Reply:
> we want to support also ubuntu, and running github actions (which we use something else)
> 
> using `subprocess.check_output` vs. `shutil.which` isn't big of a difference from my POV

i see. guess that's why you prefer using `update-alternatives --display java`.

Reply:
BTW I can confirm Cassandra 4.1 was working o.k. with jvm8, at least for our dtest needs (only running a few tests)

Reply:
> BTW I can confirm Cassandra 4.1 was working o.k. with jvm8, at least for our dtest needs (only running a few tests)

Should add cassandra 4.1 to the integration tests CI of CCM